MIT Press Journals.js

```diff
@@ -1,79 +1,123 @@
+	var rows = doc.querySelectorAll('.art_title>a.ref');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
 function doWeb(doc, url) {
-	var articles = new Array();
 	if (detectWeb(doc, url) == "multiple") {
-		var items = new Object();
-		var links = doc.evaluate('//table[@class="articleEntry"]/tbody/tr//a[text() = "First Page" or text() = "Citation" or text() = "Abstract"]', doc, null, XPathResult.ANY_TYPE, null);
-		var titles = doc.evaluate('//table[@class="articleEntry"]/tbody/tr//div[@class="arttitle"]', doc, null, XPathResult.ANY_TYPE, null);
-		var link, title;
-		while ((link = links.iterateNext()) && (title = titles.iterateNext())) {
-			items[link.href] = Zotero.Utilities.trimInternal(title.textContent);
-		}
-		
-		Zotero.selectItems(items, function (items) {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
 			if (!items) {
 				return true;
 			}
-			urls = new Array();
+			var articles = [];
 			for (var i in items) {
-				articles.push('http://www.mitpressjournals.org/doi/abs/' + getDOI(i));
+				articles.push(i);
 			}
-			ZU.processDocuments(articles, scrape)
+			ZU.processDocuments(articles, scrape);
 		});
 	} else {
-		scrape(doc, 'http://www.mitpressjournals.org/doi/abs/' + getDOI(url));
+		scrape(doc, url);
 	}
 }
 
+
 function scrape(doc, url){
-		if (doc.evaluate('//div[@class="abstractSection"]/p[contains(@class, "last") or contains(@class, "first")]', doc, null, XPathResult.ANY_TYPE, null).iterateNext()) {
-			var abs = Zotero.Utilities.trimInternal(doc.evaluate('//div[@class="abstractSection"]/p[contains(@class, "last") or contains(@class, "first")]', doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent);
-		}
-		var doi = getDOI(doc.location.href);
-		var risurl = 'http://www.mitpressjournals.org/action/downloadCitation?doi=' + doi + '&include=cit&format=refman&direct=on&submit=Download+article+metadata';		
-		var pdfurl = doc.location.href.replace("/doi/abs/", "/doi/pdf/");
-		Zotero.Utilities.HTTP.doGet(risurl, function(text) {
-			var translator = Zotero.loadTranslator("import");
-			translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
-			//Zotero.debug(text)
-			translator.setString(text);
-			translator.setHandler("itemDone", function(obj, item) {
-				//picks up some weird attachments and notes from the RIS - delete
-				item.attachments= [];
-				item.notes=[];
-				item.attachments.push({url:doc.location.href, title:item.publicationTitle + " Snapshot", mimeType:"text/html"})
-				item.attachments.push({url:pdfurl, title:item.publicationTitle + " Full Text PDF", mimeType:"application/pdf"});
-				if (abs) item.abstractNote = abs;
-				item.complete();	
+	var abs = text(doc, '.abstractSection');
+	var doi = getDOI(doc.location.href);
```

Comment:
why are you using doc.location.href instead of url here and below?

Reply:
I didn't touch these lines... I guess that we can simply use `url` instead and nothing will change. Scaffold runs for me still a little weird, such that testing is still not so comfortable, but I try to look into this soon.

Reply:
The search and replace in Scaffold is here quite handy:
![replace-scaffold](https://user-images.githubusercontent.com/5199995/32698278-07925512-c7a3-11e7-98a6-cb6405965496.png)